### PR TITLE
Various fixes for 1.6.1

### PR DESCRIPTION
**Pull request by hrydgard**
Fixes some issues from #11074 plus a couple more. Plus, adds some checks that I don't really understand how they'd fix anything but that call stacks point strongly to....

### CODE REVIEW

File: android/src/org/ppsspp/ppsspp/NativeActivity.java [hunk 1]
```diff
@@ -296,7 +296,12 @@ public void Initialize() {
 	    File sdcard = Environment.getExternalStorageDirectory();
 
 	    String externalStorageDir = sdcard.getAbsolutePath();
-	    String dataDir = this.getFilesDir().getAbsolutePath();
+	    File filesDir = this.getFilesDir();
+		String dataDir = null;
```
Comment (unknownbrackets) [inline android/src/org/ppsspp/ppsspp/NativeActivity.java]: nit: Tabs and spaces living in despicable harmony.

-[Unknown]

File: android/src/org/ppsspp/ppsspp/NativeActivity.java [hunk 2]
```diff
@@ -296,7 +296,12 @@ public void Initialize() {
...
+	    if (filesDir != null) {
+	    	// Null has been seen in Google Play stacktraces
+			dataDir = this.getFilesDir().getAbsolutePath();
```
Comment (unknownbrackets) [inline android/src/org/ppsspp/ppsspp/NativeActivity.java]: Should we use `filesDir.getAbsolutePath()`?  Not that it matters much.

Could this maybe happen when we don't have storage access, I guess?

-[Unknown]

Comment (hrydgard) [reply android/src/org/ppsspp/ppsspp/NativeActivity.java]: Ah yeah, that's nicer.

No, the datadir is a private directory that should always be accessible. All I can think of is platform bugs combined with disk full.

File: android/src/org/ppsspp/ppsspp/NativeActivity.java (hunk 1 above)
Comment (hrydgard) [reply android/src/org/ppsspp/ppsspp/NativeActivity.java]: I know :( not gonna bother fixing though, as you say we should just rip the band-aid.

**Review by unknownbrackets — APPROVED** [review]

**Pull request merged**